include/h2o/string_.h

```diff
@@ -127,7 +127,20 @@ h2o_iovec_t h2o_str_stripws(const char *s, size_t len);
  */
 size_t h2o_strstr(const char *haysack, size_t haysack_len, const char *needle, size_t needle_len);
 /**
+ * Parses a comma-separated string or a comma-separated list of sublists into tokens. Each token is returned as a tuple of
+ * (returned_pointer_being_non_null, *element_len). When all tokens are consumed, NULL is returned. See t/00unit/lib/common/string.c
+ * for examples.
  *
+ * @param iter       Iterator. When calling the function for the first time, this vector should point to the entire string to be
+ *                   parsed.
+ * @param separator  Separator to separate tokens. This value should be set to `','` when separating a simple comma-separated list.
+ *                   A different separator can be specified when parsing a comma-separated list that consists of sublists using a
+ *                   different separator (e.g., a link header is a comma-separated list of sublists using semicolons). In such case,
+ *                   this function can be used to iterate through the sublist elements. Boundary of sublists is signaled to the
+ *                   caller by returning `,`.
+ * @param value      [optional] When a non-NULL address is given and if the offending token contains `=`, the token is split into
```

Comment:
s/offending/returned/ ?

Reply:
Thank you for the review. When with "found element", as the topic is the substring that might be a returned token, or that might be a token-value pair.